### ISSUE TITLE
Add type safety and improve type inference in `UnitsMath.cs`

### ISSUE THREAD

**Issue by UrielZyx** [angularsen/UnitsNet#1373]
**Is your feature request related to a problem? Please describe.**
Some of the extension methods in`UnitsMath.cs` (e.g. `Average`) take an `Enum unitType` argument. Our system uses these extensively, and when copying/pasting code (and changing the quantity type because we're dealing with different sensors every time), I'd like the compiler to remind me I forgot to update the unit.

**Describe the solution you'd like**
Change

```C#
public static TQuantity Average<TQuantity>(
    this IEnumerable<TQuantity> source,
    Enum unitType)
    where TQuantity : IQuantity
```

into something like:

```C#
public static TQuantity Average<TQuantity>(
    this IEnumerable<TQuantity> source,
    TUnitType unitType)
    where TUnitType : Enum
    where TQuantity : IQuantity<TUnitType>
```

**Describe alternatives you've considered**
We tried implementing type safe methods in our own codebase, but when we use the wrong unit type, our IDE just falls back to the implementation in `UnitsMath`.

**Additional context**
- This will break backwards compatibility, but only for people doing really weird things.
- This change might be warranted in a lot of other places that take a `unitType` parameter (I haven't checked yet, mostly because they didn't come up in my use-case)

**Comment by UrielZyx** [comment]
Sounds great.
By "target" do you mean my branch off of `release/v6` (sorry, I have limited experience with GitHub)?
Is there a timeframe for the PR?

**Comment by UrielZyx** [comment]
I tried pushing a branch and got this:
"remote: Permission to angularsen/UnitsNet.git denied to UrielZyx."
Do I need special permissions? Also, if you get notified for each time this happens then I'm sorry for the spam (I tried multiple times, including a few that accidentally used my work account)

**Comment by UrielZyx** [comment]
Oh, looking at other PRs it seems I need to fork the repo.
I'll try and get to that later. Thanks

**Comment by UrielZyx** [comment]
PR is ready at https://github.com/angularsen/UnitsNet/pull/1374
I looked around to see if I can find other places with the same issue. The only one I could find was in `UnitConverter.cs`, but it breaks generated code, so probably better to keep for a different PR.

**Comment by angularsen** [comment]
Fixed by #1374
